[BUG] Redis 관련 환경변수 도커 컴포즈 파일에 추가 및 env 설정

### 📄 어떤 버그인가요?

docker compose 파일에 redis 관련된 환경변수가 누락되어 스프링과 redis 간 연결이 원활하게 이루어지지 않고 있습니다. 


### 🧐 재현 방법

카카오 로그인 버튼을 누르면 이런 에러가 발생합니다. 

### 🤔 예상 결과

ec2 서버에서 수동으로 docercompose 폴더에 env 설정을 추가하니 해당 에러는 사라졌습니다. 

### 📸 스크린샷 또는 로그

```
2025-10-27T21:38:10.797+09:00  WARN 1 --- [Team18_BE] [oundedElastic-1] o.s.b.a.d.r.RedisReactiveHealthIndicator : Redis health check failed

org.springframework.data.redis.RedisConnectionFailureException: Unable to connect to Redis
        at org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory$ExceptionTranslatingConnectionProvider.translateException(LettuceConnectionFactory.java:1866) ~[spring-data-redis-3.5.3.jar!/:3.5.3]
        at org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory$ExceptionTranslatingConnectionProvider.getConnection(LettuceConnectionFactory.java:1797) ~[spring-data-redis-3.5.3.jar!/:3.5.3]
        at org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory$SharedConnection.getNativeConnection(LettuceConnectionFactory.java:1594) ~[spring-data-redis-3.5.3.jar!/:3.5.3]
        at org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory$SharedConnection.lambda$getConnection$0(LettuceConnectionFactory.java:1574) ~[spring-data-redis-3.5.3.jar!/:3.5.3]
        at org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory.doInLock(LettuceConnectionFactory.java:1535) ~[spring-data-redis-3.5.3.jar!/:3.5.3]
        at org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory$SharedConnection.getConnection(LettuceConnectionFactory.java:1571) ~[spring-data-redis-3.5.3.jar!/:3.5.3]
        at org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory.getSharedReactiveConnection(LettuceConnectionFactory.java:1282) ~[spring-data-redis-3.5.3.jar!/:3.5.3]
        at org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory.getReactiveConnection(LettuceConnectionFactory.java:1157) ~[spring-data-redis-3.5.3.jar!/:3.5.3]
        at org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory.getReactiveConnection(LettuceConnectionFactory.java:121) ~[spring-data-redis-3.5.3.jar!/:3.5.3]
        at reactor.core.publisher.MonoSupplier.call(MonoSupplier.java:67) ~[reactor-core-3.7.9.jar!/:3.7.9]
        at reactor.core.publisher.FluxSubscribeOnCallable$CallableSubscribeOnSubscription.run(FluxSubscribeOnCallable.java:228) ~[reactor-core-3.7.9.jar!/:3.7.9]
        at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:68) ~[reactor-core-3.7.9.jar!/:3.7.9]
        at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:28) ~[reactor-core-3.7.9.jar!/:3.7.9]
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[na:na]
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[na:na]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[na:na]
Caused by: io.lettuce.core.RedisConnectionException: Unable to connect to localhost/<unresolved>:6379
        at io.lettuce.core.RedisConnectionException.create(RedisConnectionException.java:63) ~[lettuce-core-6.6.0.RELEASE.jar!/:6.6.0.RELEASE/643bd47]
        at io.lettuce.core.RedisConnectionException.create(RedisConnectionException.java:41) ~[lettuce-core-6.6.0.RELEASE.jar!/:6.6.0.RELEASE/643bd47]
        at io.lettuce.core.AbstractRedisClient.getConnection(AbstractRedisClient.java:354) ~[lettuce-core-6.6.0.RELEASE.jar!/:6.6.0.RELEASE/643bd47]
        at io.lettuce.core.RedisClient.connect(RedisClient.java:220) ~[lettuce-core-6.6.0.RELEASE.jar!/:6.6.0.RELEASE/643bd47]
        at org.springframework.data.redis.connection.lettuce.StandaloneConnectionProvider.lambda$getConnection$1(StandaloneConnectionProvider.java:112) ~[spring-data-redis-3.5.3.jar!/:3.5.3]
        at java.base/java.util.Optional.orElseGet(Optional.java:364) ~[na:na]
        at org.springframework.data.redis.connection.lettuce.StandaloneConnectionProvider.getConnection(StandaloneConnectionProvider.java:112) ~[spring-data-redis-3.5.3.jar!/:3.5.3]
        at org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory$ExceptionTranslatingConnectionProvider.getConnection(LettuceConnectionFactory.java:1795) ~[spring-data-redis-3.5.3.jar!/:3.5.3]
        ... 16 common frames omitted
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:6379
Caused by: java.net.ConnectException: Connection refused
        at java.base/sun.nio.ch.Net.pollConnect(Native Method) ~[na:na]
        at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:682) ~[na:na]
        at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:973) ~[na:na]
        at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:336) ~[netty-transport-4.1.124.Final.jar!/:4.1.124.Final]
        at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:339) ~[netty-transport-4.1.124.Final.jar!/:4.1.124.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:784) ~[netty-transport-4.1.124.Final.jar!/:4.1.124.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:732) ~[netty-transport-4.1.124.Final.jar!/:4.1.124.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:658) ~[netty-transport-4.1.124.Final.jar!/:4.1.124.Final]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[netty-transport-4.1.124.Final.jar!/:4.1.124.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998) ~[netty-common-4.1.124.Final.jar!/:4.1.124.Final]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.124.Final.jar!/:4.1.124.Final]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.124.Final.jar!/:4.1.124.Final]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[na:na]
```

--- a/.github/workflows/deploy-with-docker.yml
+++ b/.github/workflows/deploy-with-docker.yml
@@ -158,7 +158,7 @@ jobs:
 
             # .env 생성 (printf로 안전하게 생성)
             tmp_env="$(mktemp)"
-            printf 'DB_NAME=%s\nDB_USER=%s\nDB_PASSWORD=%s\nDB_ROOT_PASSWORD=%s\nECR_REGISTRY=%s\nKAKAO_CLIENT_ID=%s\nKAKAO_CLIENT_SECRET=%s\nJWT_SECRET=%s\nKAKAO_REDIRECT_URI_PROD=%s\nSMTP_HOST=%s\nSMTP_PORT=%s\nSMTP_USERNAME=%s\nSMTP_PASSWORD=%s\nEMAIL_FROM=%s\nEMAIL_SUBJECT_PREFIX=%s\n' \
+            printf 'DB_NAME=%s\nDB_USER=%s\nDB_PASSWORD=%s\nDB_ROOT_PASSWORD=%s\nECR_REGISTRY=%s\nKAKAO_CLIENT_ID=%s\nKAKAO_CLIENT_SECRET=%s\nJWT_SECRET=%s\nKAKAO_REDIRECT_URI_PROD=%s\nSMTP_HOST=%s\nSMTP_PORT=%s\nSMTP_USERNAME=%s\nSMTP_PASSWORD=%s\nEMAIL_FROM=%s\nEMAIL_SUBJECT_PREFIX=%s\nREDIS_HOST=%s\nREDIS_PORT=%s\n' \
             '${{ secrets.DB_NAME }}' \
             '${{ secrets.DB_USER }}' \
             '${{ secrets.DB_PASSWORD }}' \
@@ -173,7 +173,9 @@ jobs:
             '${{ secrets.SMTP_USERNAME }}' \
             '${{ secrets.SMTP_PASSWORD }}' \
             '${{ secrets.EMAIL_FROM }}' \
-            '${{ secrets.EMAIL_SUBJECT_PREFIX }}' > "$tmp_env"
+            '${{ secrets.EMAIL_SUBJECT_PREFIX }}' \
+            '${{ secrets.REDIS_HOST }}' \
+            '${{ secrets.REDIS_PORT }}' > "$tmp_env"
             install -m 600 "$tmp_env" ${{ env.REMOTE_DIR }}/shared/.env
             rm -f "$tmp_env"
             ln -sf ${{ env.REMOTE_DIR }}/shared/.env ${{ env.REMOTE_DIR }}/current/.env

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -80,6 +80,10 @@ services:
       EMAIL_FROM: ${EMAIL_FROM}
       SPRING_EMAIL_SUBJECT_PREFIX: ${EMAIL_SUBJECT_PREFIX}
 
+      # Redis
+      SPRING_DATA_REDIS_HOST: ${REDIS_HOST}
+      SPRING_DATA_REDIS_PORT: ${REDIS_PORT}
+
     healthcheck:
       # 상태 코드 200이고 본문에 "status":"UP" 포함되면 성공
       test: [ "CMD-SHELL", "wget -qO- http://localhost:8080/actuator/health | grep -q '\"status\":\"UP\"'" ]

--- a/src/main/java/com/kakaotech/team18/backend_server/domain/clubApplyForm/dto/ClubApplyFormRequestDto.java
+++ b/src/main/java/com/kakaotech/team18/backend_server/domain/clubApplyForm/dto/ClubApplyFormRequestDto.java
@@ -1,5 +1,6 @@
 package com.kakaotech.team18.backend_server.domain.clubApplyForm.dto;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.kakaotech.team18.backend_server.domain.formQuestion.dto.FormQuestionRequestDto;
 import io.swagger.v3.oas.annotations.media.Schema;
 import jakarta.validation.Valid;
@@ -37,6 +38,8 @@ public record ClubApplyFormRequestDto(
         List<@Valid FormQuestionRequestDto> formQuestions
 ) {
         @AssertTrue(message = "모집 마감일은 시작일과 같거나 이후여야 합니다.")
+        @JsonIgnore
+        @Schema(hidden = true)
         public boolean isRecruitPeriodValid() {
                 return recruitStart == null || recruitEnd == null || !recruitEnd.isBefore(recruitStart);
         }

--- a/src/main/java/com/kakaotech/team18/backend_server/domain/clubApplyForm/dto/ClubApplyFormUpdateDto.java
+++ b/src/main/java/com/kakaotech/team18/backend_server/domain/clubApplyForm/dto/ClubApplyFormUpdateDto.java
@@ -1,5 +1,6 @@
 package com.kakaotech.team18.backend_server.domain.clubApplyForm.dto;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.kakaotech.team18.backend_server.domain.formQuestion.dto.FormQuestionUpdateDto;
 import io.swagger.v3.oas.annotations.media.Schema;
 import jakarta.validation.Valid;
@@ -37,6 +38,8 @@ public record ClubApplyFormUpdateDto(
         List<@Valid FormQuestionUpdateDto> formQuestions
 ) {
         @AssertTrue(message = "모집 마감일은 시작일과 같거나 이후여야 합니다.")
+        @JsonIgnore
+        @Schema(hidden = true)
         public boolean isRecruitPeriodValid() {
                 return recruitStart == null || recruitEnd == null || !recruitEnd.isBefore(recruitStart);
         }

